src/util/domains.py

```diff
@@ -196,12 +196,12 @@ def _adapt_one_domain(cr, target_model, old, new, model, domain, adapter=None, f
     if not adapter:
         adapter = lambda leaf, _, __: [leaf]
 
-    evaluation_context = SelfPrintEvalContext()
-
     # pre-check domain
     if isinstance(domain, basestring):
+        replaced_domain, ctx = SelfPrintEvalContext.preprocess(domain)
         try:
```

Comment:
It could be placed in the try/except
```suggestion
        try:
            replaced_domain, ctx = SelfPrintEvalContext.preprocess(domain)
```

Reply:
> It could be placed in the try/except
Actually, it MUST to catch SyntaxError...

Reply:
I pushed a try/except internal to preprocess. Also saquashed commits.